modules/openapi-generator/src/main/resources/elixir/request_builder.ex.mustache

```diff
@@ -107,20 +107,22 @@ defmodule {{moduleName}}.RequestBuilder do
 
   ## Parameters
 
-  - arg1 (Tesla.Env.t | term) - The response object
+  - arg1 ({:ok, Tesla.Env.t} | term) - The response object
   - arg2 (:false | struct | [struct]) - The shape of the struct to deserialize into
 
   ## Returns
 
   {:ok, struct} on success
   {:error, term} on failure
   """
-  @spec decode(Tesla.Env.t | term()) :: {:ok, struct()} | {:error, Tesla.Env.t} | {:error, term()}
-  def decode(%Tesla.Env{status: 200, body: body}), do: Poison.decode(body)
+  @spec decode({:ok, Tesla.Env.t} | term()) :: {:ok, struct()} | {:error, Tesla.Env.t} | {:error, term()}
+  def decode({:ok, %Tesla.Env{status: 200, body: body}}), do: Poison.decode(body)
   def decode(response), do: {:error, response}
```

Comment:
The current `response` matches on all responses whicht didn't result in a 200 status code (which might needs to be fixed in another Issue - I'll open that one later)

Therefore I would suggest something like this:
```suggestion
   def decode({:ok, %Tesla.Env{} = response), do: {:error, response}
```

Reply:
I just created a issue for this bug, so fixing the hardcoded `200` match doesn't have to be fixed in this PR :) https://github.com/OpenAPITools/openapi-generator/issues/2334

Reply:
@mrmstn Sounds good, we can left that for a different pr